platform/wm/rhodes/browser/CEBrowserEngine.cpp

```diff
@@ -1157,4 +1164,134 @@ BOOL CEBrowserEngine::ZoomTextOnTab(int nZoom, UINT iTab)
 }
 
 
+DWORD WINAPI CEBrowserEngine::NetworkWindowThread( LPVOID lpParameter )
+{
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Started";
+	CEBrowserEngine * pEng = (CEBrowserEngine*) lpParameter;
+
+	TCHAR szWindowText[MAX_PATH];
+	TCHAR szWindowClass[MAX_PATH];
+	HWND hCurrentWindow = NULL;
+
+	ZeroMemory(szWindowText,MAX_PATH);
+	ZeroMemory(szWindowClass,MAX_PATH);
+
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread";
+	LOG(INFO) + pEng->m_bNavigationComplete;
+	
+	while(!pEng->m_bNavigationComplete)
+	{
+		HWND hCurrentWindow=FindWindow(TEXT("Dialog"),TEXT("Enter Network Password"));
+		if(hCurrentWindow != NULL)
+		{
+
+#ifdef _DEBUG
+			GetWindowText(hCurrentWindow,szWindowText,MAX_PATH);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  Title";
+			LOG(INFO) + szWindowText;
+			ZeroMemory(szWindowClass,MAX_PATH);
+			GetClassName(hCurrentWindow,szWindowClass,MAX_PATH);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  class";
+			LOG(INFO) + szWindowClass;
+			LONG lIndex = GetWindowLong(hCurrentWindow,GWL_ID);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread FindWindow  id";
+			LOG(INFO) + lIndex;
+#endif
+			::EnumChildWindows(hCurrentWindow,(WNDENUMPROC)NetworkWndProc,0);
+			LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended (0)";
+			pEng->m_bNavigationComplete = TRUE;
+			return 0;	
+		}
+		
+	}
+
+	LOG(INFO) + "CEBrowserEngine::NetworkWindowThread Ended";
+	pEng->m_bNavigationComplete = TRUE;
+	return 0;
+}
+
+// EMBPD00158491 - [SAP-ITS][CE5/MK4000]-Sip is not shown while trying to enter text in fields of Authentication screen
+// SIP not appearing on authentication window on IE engine has been fixed
+LRESULT CALLBACK NetworkWndProc(HWND hWnd,LPARAM lParam)
+{
+#ifdef _DEBUG
+	LOG(INFO) + "NetworkWndProc" ;
+#endif
+	if(hWnd)
+	{
+
+		TCHAR szChildWndText[MAX_PATH];
+		TCHAR szChildWndClass[MAX_PATH];
+
+		ZeroMemory(szChildWndText,MAX_PATH);
+		ZeroMemory(szChildWndClass,MAX_PATH);
+
+		GetWindowText(hWnd,szChildWndText,MAX_PATH);
+		GetClassName(hWnd,szChildWndClass,MAX_PATH);
+#ifdef _DEBUG
+		LOG(INFO) + "NetworkWndProc Window text";
+		LOG(INFO) + szChildWndText;
+
+		LOG(INFO) + "NetworkWndProc Window class";
+		LOG(INFO) + szChildWndClass;
+
+
+		LOG(INFO) + "NetworkWndProc Window class";
+		LONG lIndex = GetWindowLong(hWnd,GWL_ID);
+		LOG(INFO) + lIndex;
+#endif
+		if(_tcsicmp(szChildWndClass,TEXT("Edit")) ==0)
+		{
+			g_wpEditWndProc= (WNDPROC)SetWindowLong(hWnd,GWL_WNDPROC,(LONG)EditWndProc);
+			if(g_wpEditWndProc== NULL)
+			{
+				LOG(ERROR) + "[NetworkWndProc] Setwindowlong returns null";
+				LOG(ERROR) + GetLastError();
+			}
+			else
+			{
+				LOG(INFO) + "[NetworkWndProc] Setwindowlong returns";
+				LOG(INFO) + GetLastError();
+			}
```

Comment:
Also, please remember that any logs are user facing, especially error messages should be useful to the user of the software.